tests/data/modules/plugin_native_types/plugins/__init__.py

```diff
@@ -32,3 +34,16 @@ def many_arguments(il: list[str], idx: int) -> str:
 @plugin
 def as_none(value: str) -> None:
     pass
+
+
+# Annotated values
+
+
+@plugin
+def annotated_arg_entity(value: Annotated[object, plugin_typing.InmantaType("TestEntity")]) -> None:
```

Comment:
Could you also add one that uses a `typedef` as its dsl type? e.g. a toy example of a valid use case might be

DSL: `typedef response as string matching self in ["yes", "no"]`
Python:
```python
@plugin
def process_response(response: Annotated[Literal["yes", "no"], InmantaType("response")]) -> None ...
```